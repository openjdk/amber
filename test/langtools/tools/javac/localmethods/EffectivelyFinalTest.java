/*
 * @test /nodynamiccopyright/
 * @bug 8233038
 * @summary Tests need to be written for local methods feature
 * @compile/fail/ref=EffectivelyFinalTest.out -XDrawDiagnostics EffectivelyFinalTest.java
 */

public class EffectivelyFinalTest {
    {
        int x = 10;
        x = 30;
        void foo() {
            x = 40;
        }
    }
    public static void main(String args []) {
       void foo() {
           args = null;
       }
       int p;
       if (args == null)
           p = 10;
       else
           p = 10;
       int goo() {
           return p;
       }
       int q;
       if (args == null)
           q = 10;
       else
           q = 10;
       int qoo() {
           return q;
       }
       q = 50;
    }
}
