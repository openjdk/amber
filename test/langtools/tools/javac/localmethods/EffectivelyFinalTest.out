EffectivelyFinalTest.java:13:13: compiler.err.cant.ref.non.effectively.final.var: x, (compiler.misc.local.meth)
EffectivelyFinalTest.java:18:12: compiler.err.cant.ref.non.effectively.final.var: args, (compiler.misc.local.meth)
EffectivelyFinalTest.java:34:19: compiler.err.cant.ref.non.effectively.final.var: q, (compiler.misc.local.meth)
3 errors
